app/spaces/[spaceid]/trustful/my-badges/details/components/MyBadgeDetails.tsx

```diff
@@ -0,0 +1,426 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { CheckIcon, CloseIcon } from '@chakra-ui/icons';
+import {
+  Box,
+  Button,
+  Card,
+  Flex,
+  Text,
+  Divider,
+  Textarea,
+} from '@chakra-ui/react';
+import { useParams, useRouter } from 'next/navigation';
+import { BeatLoader } from 'react-spinners';
+import { encodeAbiParameters, parseAbiParameters } from 'viem';
+import { scroll, scrollSepolia } from 'viem/chains';
+import { useAccount, useSwitchChain } from 'wagmi';
+import { useTrustful } from '@/context/TrustfulContext';
+import {
+  BadgeStatus,
+  BadgeTagIcon,
+} from '@/app/spaces/[spaceid]/trustful/my-badges/components';
+import { getEllipsedAddress } from '@/utils/format';
+import {
+  isDev,
+  TRUSTFUL_SCHEMAS,
+} from '@/app/spaces/[spaceid]/trustful/constants/constants';
+import {
+  AttestationRequestData,
+  submitAttest,
+  revoke,
+} from '@/app/spaces/[spaceid]/trustful/service';
+import { HeartLoveIcon, UserIcon } from '@/components/icons';
+import {
+  BadgeDetailsNavigation,
+  CopyToClipboardButton,
+  OutboundLinkButton,
+  TheHeader,
+  TheFooterNavbar,
+} from '@/app/spaces/[spaceid]/trustful/components';
+import toast from 'react-hot-toast';
+import { Address } from 'viem';
+
+export const MyBadgeDetails = () => {
+  const { address, chainId } = useAccount();
+  const { setSelectedBadge, selectedBadge } = useTrustful();
+  const { push } = useRouter();
+  const params = useParams();
+
+  // const { villagerAttestationCount } = useContext(WalletContext);
+  const villagerAttestationCount = Number(1);
```

Comment:
Please add comment on this and I assume this will be dynamic, right?

```suggestion
  // const { villagerAttestationCount } = useContext(WalletContext);
  const villagerAttestationCount = Number(1);
```